### PR TITLE
Wsl systemd

### PR DESCRIPTION
**Pull request by jreidinger**
## Problem

WSL now supports systemd when some tuning is done. And that tuning is packaged as pattern with bunch of deps.

jira: https://jira.suse.com/browse/PED-5099

## Solution

Implement checkbox as requested and also adapt a bit help text as mentioned in that jira entry.

## Testing

- *Added a new unit test*
- Manually tested and documented how to test it.

## Screenshot

wsl_firstboot

wsl_firstboot_help

### CODE REVIEW

File: src/lib/y2firstboot/dialogs/wsl_product_selection.rb [hunk 1]
```diff
@@ -87,17 +95,30 @@ def dialog_content
             # TRANSLATORS: check box label
             Left(CheckBox(Id(:wsl_gui_pattern),
               _("Install WSL GUI pattern (requires registration)"),
-              wsl_gui_pattern))
+              wsl_gui_pattern)),
+            # TRANSLATORS:
+            Label(_("The WSL systemd pattern provides wsl.conf adjustment and init symlink for\n" \
```
Comment (lslezak) [inline src/lib/y2firstboot/dialogs/wsl_product_selection.rb:100]: The checkbox label is IMHO too technical, that should be part of the documentation and in the help text.

For label I'd prefer some simpler, less technical description understandable for more users.

Something like "The WSL systemd pattern allows using the Linux services inside a WSL container."

Comment (jreidinger) [reply src/lib/y2firstboot/dialogs/wsl_product_selection.rb:100]: it is kind of similar to the one about WSL GUI. I am fine with simplifying both, but I am not sure about who is target user.

File: src/lib/y2firstboot/dialogs/wsl_product_selection.rb [hunk 2]
```diff
@@ -81,24 +91,40 @@ def dialog_content
...
-              "In that case the system needs to be registered as well.</p>")
+          # TRANSLATORS: help text (2/3)
+          _("<p>For smoother experience with graphical programs in WSL " \
+              "the WSL GUI pattern provides recommended config, tools and libraries. " \
+              "In that case the system needs to be registered as well.</p>") +
+          # TRANSLATORS: help text (3/3)
+          _("<p>For enablement of systemd in WSL the WSL systemd pattern provides wsl.conf " \
+              "and /sbin/init adjustments. " \
+              "In that case the system needs to be registered as well. " \
+              "Also be aware that systemd enablement is in effect only after relaunch.</p>")
```
Comment (joseivanlopez) [inline src/lib/y2firstboot/dialogs/wsl_product_selection.rb:127]: I would reword the last sentence a bit: "Reboot is required for enabling systemd".

File: src/lib/y2firstboot/dialogs/wsl_product_selection.rb (hunk 1 above)
Comment (joseivanlopez) [reply src/lib/y2firstboot/dialogs/wsl_product_selection.rb]: I agree with @lslezak. I would something like he proposed.

File: src/lib/y2firstboot/dialogs/wsl_product_selection.rb (hunk 2 above)
Comment (jreidinger) [reply src/lib/y2firstboot/dialogs/wsl_product_selection.rb:127]: problem is that in wsl it is not reboot. That is why I use relaunch...so I would use your sentence with Relaunch

File: src/lib/y2firstboot/dialogs/wsl_product_selection.rb (hunk 1 above)
Comment (jreidinger) [reply src/lib/y2firstboot/dialogs/wsl_product_selection.rb]: well, goal of the PBI is to use similar wording like in pattern description and I also ask on discuss-wsl on slack for feedback. To be honest I am not sure about terms used in WSL like if we should talk about services or container in WSL context.

File: doc/testing_wsl.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+### Creating Testing APPX
+
+The easiest way for me is to use OBS. Find the target release for WSL image like `SUSE:SLE-15-SP6:Update:WSL`
+and branch there `kiwi-images-wsl` that is responsible for creation of appx. In your branch create package
+yast2-firstboot ( as it is usually inherited from target and does not live in WSL subproject ) and copy there
+modified sources with rake tarball. It is also mandatory to link or branch package `wsl-appx` as it ensures
+that certificate used for appx matches with the one used in kiwi.
+Another important think is to ensure that certificate for home repo is valid and not outdated. To check it use
+`osc signkey --sslcert home:...` and if it is invalid use `osc signkey --sslcert --create home:jreidinger`.
```
[the remark this answers is not in the document]
Comment (jreidinger) [reply doc/testing_wsl.md:14]: I will use similar like in previous command that `home:...` to be consistent

File: doc/testing_wsl.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+### Installing Modified APPX
+
+Just follow instructions on https://en.opensuse.org/WSL/Manual_Installation
```
[the remark this answers is not in the document]
Comment (jreidinger) [reply doc/testing_wsl.md:18]: yeah, I plan to use "on page" and ends in the middle.

**Review by joseivanlopez — APPROVED** [review]

File: src/lib/y2firstboot/dialogs/wsl_product_selection.rb (hunk 1 above)
Comment (joseivanlopez) [reply src/lib/y2firstboot/dialogs/wsl_product_selection.rb]: Ok, well, we can change it if somebody complains.
